<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1020</width>
    <height>623</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Демонстрація роботи алгоритмів</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGraphicsView" name="graphicsView">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>210</y>
      <width>620</width>
      <height>320</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Тут у вигляді діаграми демонструються елементи структури&lt;/p&gt;&lt;p&gt;в процесі виконання алгоритму, де їхня висота пропорційна величині, що зберігає елемент.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_algs">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>331</width>
      <height>81</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>301</width>
       <height>55</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLabel" name="label_alg">
        <property name="text">
         <string>Алгоритм, що буде демонструватись</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="comboBox">
        <item>
         <property name="text">
          <string>&lt;оберіть алгоритм&gt;</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Bubble Sort</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_str">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>331</width>
      <height>291</height>
     </rect>
    </property>
    <property name="title">
     <string>Структура</string>
    </property>
    <widget class="QRadioButton" name="radioButton_rnd">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>311</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Випадкова генерація</string>
     </property>
    </widget>
    <widget class="QGroupBox" name="groupBox_rnd">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>311</width>
       <height>141</height>
      </rect>
     </property>
     <property name="title">
      <string/>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>11</y>
        <width>291</width>
        <height>92</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label_amt">
           <property name="text">
            <string>Кількість елементів</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_amt"/>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="label_interv">
         <property name="text">
          <string>Інтервали:</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_min">
           <property name="text">
            <string>Мін.</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_min"/>
         </item>
         <item>
          <widget class="QLabel" name="label_max">
           <property name="text">
            <string>Макс.</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_max"/>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QPushButton" name="pushButton_rnd">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>105</y>
        <width>141</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Ввести</string>
      </property>
     </widget>
    </widget>
    <widget class="QRadioButton" name="radioButton_file">
     <property name="geometry">
      <rect>
       <x>11</x>
       <y>212</y>
       <width>152</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Введення із файлу</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_file">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>210</y>
       <width>151</width>
       <height>29</height>
      </rect>
     </property>
     <property name="text">
      <string>Обрати файл</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="radioButton_man">
     <property name="geometry">
      <rect>
       <x>12</x>
       <y>252</y>
       <width>143</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Введення вручну</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_man">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>250</y>
       <width>151</width>
       <height>29</height>
      </rect>
     </property>
     <property name="text">
      <string>Ввести</string>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="label_str_entered">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>360</y>
      <width>331</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QGraphicsView" name="graphicsView_2">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>40</y>
      <width>621</width>
      <height>141</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Структура&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>190</y>
      <width>211</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Демонстрація алгоритму</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>20</y>
      <width>91</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Структура</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_print">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>410</y>
      <width>93</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>PushButton</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="plainTextEdit">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>390</y>
      <width>171</width>
      <height>87</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1020</width>
     <height>17</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_file">
    <property name="title">
     <string>Файл</string>
    </property>
    <addaction name="separator"/>
    <addaction name="action_exit"/>
   </widget>
   <widget class="QMenu" name="menu_demo">
    <property name="title">
     <string>Демонстрація</string>
    </property>
    <widget class="QMenu" name="menu_3">
     <property name="title">
      <string>Алгоритми</string>
     </property>
    </widget>
    <widget class="QMenu" name="menu_4">
     <property name="title">
      <string>Історія</string>
     </property>
     <addaction name="separator"/>
    </widget>
    <addaction name="menu_3"/>
    <addaction name="separator"/>
    <addaction name="menu_4"/>
    <addaction name="separator"/>
    <addaction name="action_start_auto"/>
    <addaction name="action_pause"/>
    <addaction name="separator"/>
    <addaction name="action_manual"/>
    <addaction name="action_prev_step"/>
    <addaction name="action_next_step"/>
    <addaction name="separator"/>
    <addaction name="action_stop"/>
   </widget>
   <addaction name="menu_file"/>
   <addaction name="menu_demo"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_start_auto"/>
   <addaction name="action_pause"/>
   <addaction name="separator"/>
   <addaction name="action_manual"/>
   <addaction name="action_prev_step"/>
   <addaction name="action_next_step"/>
   <addaction name="separator"/>
   <addaction name="action_stop"/>
  </widget>
  <action name="action_exit">
   <property name="text">
    <string>Вийти</string>
   </property>
   <property name="toolTip">
    <string>Вийти із програми</string>
   </property>
  </action>
  <action name="action_start_auto">
   <property name="icon">
    <iconset resource="img.qrc">
     <normaloff>:/img/img/start.png</normaloff>:/img/img/start.png</iconset>
   </property>
   <property name="text">
    <string>Старт (автоматично)</string>
   </property>
  </action>
  <action name="action_pause">
   <property name="icon">
    <iconset resource="img.qrc">
     <normaloff>:/img/img/pause.png</normaloff>:/img/img/pause.png</iconset>
   </property>
   <property name="text">
    <string>Пауза/продовжити</string>
   </property>
  </action>
  <action name="action_stop">
   <property name="icon">
    <iconset resource="img.qrc">
     <normaloff>:/img/img/stop.png</normaloff>:/img/img/stop.png</iconset>
   </property>
   <property name="text">
    <string>Стоп</string>
   </property>
  </action>
  <action name="action_manual">
   <property name="icon">
    <iconset resource="img.qrc">
     <normaloff>:/img/img/start_m.png</normaloff>:/img/img/start_m.png</iconset>
   </property>
   <property name="text">
    <string>Старт (вручну)</string>
   </property>
  </action>
  <action name="action_prev_step">
   <property name="icon">
    <iconset resource="img.qrc">
     <normaloff>:/img/img/prev.png</normaloff>:/img/img/prev.png</iconset>
   </property>
   <property name="text">
    <string>Попередній крок</string>
   </property>
  </action>
  <action name="action_next_step">
   <property name="icon">
    <iconset resource="img.qrc">
     <normaloff>:/img/img/next.png</normaloff>:/img/img/next.png</iconset>
   </property>
   <property name="text">
    <string>Наступний крок</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="img.qrc"/>
 </resources>
 <connections/>
</ui>
